src/geotiffimage.js

```diff
@@ -475,9 +475,7 @@ class GeoTIFFImage {
           if (this.planarConfiguration === 2) {
             bytesPerPixel = this.getSampleByteSize(sampleIndex);
           }
-          const promise = this.getTileOrStrip(xTile, yTile, sample, poolOrDecoder, signal);
-          promises.push(promise);
-          promise.then((tile) => {
```

Comment:
`Promise.then` returns a _new_ promise, which I'm guessing is what we actually want to await with `Promise.all`.